File: src/ros2_bridge/mappers/geometry_msgs/inertia_stamped.rs
//! Typed mapper for `geometry_msgs/msg/InertiaStamped`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn inertia_stamped_to_bus(msg: ros_env::geometry_msgs::msg::InertiaStamped) -> crate::geometry_msgs::msg::v1::InertiaStamped {
    crate::geometry_msgs::msg::v1::InertiaStamped {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        inertia: Some(crate::ros2_bridge::mappers::geometry_msgs::inertia::inertia_to_bus(msg.inertia)),
    }
}

pub(crate) fn inertia_stamped_to_ros(bus: crate::geometry_msgs::msg::v1::InertiaStamped) -> ros_env::geometry_msgs::msg::InertiaStamped {
    ros_env::geometry_msgs::msg::InertiaStamped {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(bus.header.unwrap_or_default()),
        inertia: crate::ros2_bridge::mappers::geometry_msgs::inertia::inertia_to_ros(bus.inertia.unwrap_or_default()),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsInertiaStampedMapper;

impl TypedTopicMapper for GeometryMsgsInertiaStampedMapper {
    type Ros = ros_env::geometry_msgs::msg::InertiaStamped;
    type Bus = crate::geometry_msgs::msg::v1::InertiaStamped;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(inertia_stamped_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(inertia_stamped_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/inertia_stamped.py
"""Generated mapper for `geometry_msgs/msg/InertiaStamped`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros
from robot_bus.ros2_bridge.mappers.geometry_msgs.inertia import inertia_to_bus, inertia_to_ros

def inertia_stamped_to_bus(msg):
    from robot_bus.geometry_msgs.msg.v1 import InertiaStamped as BusMsg

    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.inertia.CopyFrom(inertia_to_bus(msg.inertia))
    return bus


def inertia_stamped_to_ros(bus):
    from geometry_msgs.msg import InertiaStamped as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.inertia = inertia_to_ros(bus.inertia)
    return out


class GeometryMsgsInertiaStampedMapper:
    def type_name(self) -> str:
        return "geometry_msgs/msg/InertiaStamped"

    def ros_msg_type(self):
        from geometry_msgs.msg import InertiaStamped as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return inertia_stamped_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.geometry_msgs.msg.v1 import InertiaStamped as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return inertia_stamped_to_ros(bus)
